src/runtime/TemporalObject.cpp

```diff
@@ -602,6 +602,102 @@ ValueVector TemporalCalendar::calendarFields(ExecutionState& state, const Value&
     return IteratorObject::iterableToListOfType(state, fieldsArray, new ASCIIString("String"));
 }
 
+Value TemporalCalendar::calendarYear(ExecutionState& state, Object* calendar, const Value& dateLike)
+{
+    Value year = calendar->get(state, state.context()->staticStrings().lazyYear()).value(state, calendar);
+    Value result = Object::call(state, year, calendar, 1, const_cast<Value*>(&dateLike));
+
+    if (result.isUndefined()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Undefined year");
+    }
+
+    return result;
+}
+
+Value TemporalCalendar::calendarMonth(ExecutionState& state, Object* calendar, const Value& dateLike)
+{
+    Value month = calendar->get(state, state.context()->staticStrings().lazyMonth()).value(state, calendar);
+    Value result = Object::call(state, month, calendar, 1, const_cast<Value*>(&dateLike));
+
+    if (result.isUndefined()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Undefined month");
+    }
+
+    return result;
+}
+
+Value TemporalCalendar::calendarMonthCode(ExecutionState& state, Object* calendar, const Value& dateLike)
+{
+    Value monthCode = calendar->get(state, state.context()->staticStrings().lazymonthCode()).value(state, calendar);
+    Value result = Object::call(state, monthCode, calendar, 1, const_cast<Value*>(&dateLike));
+
+    if (result.isUndefined()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Undefined monthCode");
+    }
+
+    return result;
```

Comment:
There are three functions, which has a similar end from `Value result = Object::call`. These could be also moved to a helper function.